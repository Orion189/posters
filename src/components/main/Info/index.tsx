import { FC, memo } from 'react';
import { Box } from '@mui/system';
import { useTranslation } from 'react-i18next';
import Typography from '@mui/material/Typography';
import { useMovieInfo } from '@components/hooks/useMovieInfo';
import List from '@mui/material/List';
import ListItem from '@mui/material/ListItem';
import ListItemText from '@mui/material/ListItemText';
import styles from './style.module.scss';

const Info: FC = () => {
    const { t } = useTranslation();
    const { movieInfo } = useMovieInfo();

    return movieInfo ? (
        <Box className={styles.cont}>
            <Typography variant="h5" gutterBottom className={styles.panelTitle}>
                {t('components.main.Info.title')}
            </Typography>
            <List className={styles.list}>
                <ListItem>
                    <ListItemText
                        primary={t('components.main.Info.label.originalTitle')}
                        primaryTypographyProps={{
                            className: styles.listItemKey
                        }}
                        secondary={movieInfo.original_title}
                        secondaryTypographyProps={{
                            variant: 'h6',
                            className: styles.listItemValue
                        }}
                    />
                </ListItem>
                <ListItem>
                    <ListItemText
                        primary={t('components.main.Info.label.year')}
                        primaryTypographyProps={{
                            className: styles.listItemKey
                        }}
                        secondary={movieInfo.year}
                        secondaryTypographyProps={{
                            variant: 'h6',
                            className: styles.listItemValue
                        }}
                    />
                </ListItem>
                <ListItem>
                    <ListItemText
                        primary={t('components.main.Info.label.country')}
                        primaryTypographyProps={{
                            className: styles.listItemKey
                        }}
                        secondary={movieInfo.country}
                        secondaryTypographyProps={{
                            variant: 'h6',
                            className: styles.listItemValue
                        }}
                    />
                </ListItem>
                {movieInfo.director ? (
                    <ListItem>
                        <ListItemText
                            primary={t('components.main.Info.label.director')}
                            primaryTypographyProps={{
                                className: styles.listItemKey
                            }}
                            secondary={movieInfo.director}
                            secondaryTypographyProps={{
                                variant: 'h6',
                                className: styles.listItemValue
                            }}
                        />
                    </ListItem>
                ) : null}
                <ListItem>
                    <ListItemText
                        primary={t('components.main.Info.label.genre')}
                        primaryTypographyProps={{
                            className: styles.listItemKey
                        }}
                        secondary={movieInfo.genre}
                        secondaryTypographyProps={{
                            variant: 'h6',
                            className: styles.listItemValue
                        }}
                    />
                </ListItem>
                {movieInfo.runtime > 0 ? (
                    <ListItem>
                        <ListItemText
                            primary={t('components.main.Info.label.runtime')}
                            primaryTypographyProps={{
                                className: styles.listItemKey
                            }}
                            secondary={movieInfo.runtime}
                            secondaryTypographyProps={{
                                variant: 'h6',
                                className: styles.listItemValue
                            }}
                        />
                    </ListItem>
                ) : null}
                {movieInfo.summary ? (
                    <ListItem>
                        <ListItemText
                            primary={t('components.main.Info.label.summary')}
                            primaryTypographyProps={{
                                className: styles.listItemKey
                            }}
                            secondary={movieInfo.summary}
                            secondaryTypographyProps={{
                                variant: 'h6',
                                className: styles.listItemValue
                            }}
                        />
                    </ListItem>
                ) : null}
            </List>
        </Box>
    ) : null;
};

export default memo(Info);
