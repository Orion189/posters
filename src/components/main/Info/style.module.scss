@import '@themes/basic/colors.module.scss';

.cont {
    padding: 2.5rem;
}

.panelTitle {
    color: $white100;
    margin-bottom: 4rem;
}

.list {
    width: 100%;
}

.listItemKey {
    color: $white60
}

.listItemValue {
    color: $primaryColorMain100
}