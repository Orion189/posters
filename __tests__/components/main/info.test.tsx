import { cleanup, render, screen } from '@testing-library/react';
import Info from '@components/main/Info';

const movieInfo = {
    original_title: 'Test movie title',
    year: 1999,
    country: 'US',
    director: 'John Doe',
    genre: 'Drama',
    runtime: 180,
    summary: 'Test movie summary'
};

jest.mock('react-i18next', () => ({
    useTranslation: () => ({
        t: jest.fn((str) => str),
        i18n: {
            changeLanguage: () => new Promise(() => {})
        }
    })
}));

jest.mock('@components/hooks/useMovieInfo', () => ({
    useMovieInfo: () => ({
        movieInfo
    })
}));

describe('Info', () => {
    afterEach(cleanup);

    it('renders <Info /> title', () => {
        const panelTitleKey = 'components.main.Info.title';

        render(<Info />);

        const heading = screen.getByRole('heading', {
            name: panelTitleKey
        });

        expect(heading).toBeInTheDocument();
    });
    it('renders <Info /> props', () => {
        render(<Info />);

        const original_title = screen.getByText(movieInfo.original_title);
        const year = screen.getByText(movieInfo.year);
        const country = screen.getByText(movieInfo.country);
        const director = screen.getByText(movieInfo.director);
        const genre = screen.getByText(movieInfo.genre);
        const runtime = screen.getByText(movieInfo.runtime);
        const summary = screen.getByText(movieInfo.summary);

        expect(original_title).toBeInTheDocument();
        expect(year).toBeInTheDocument();
        expect(country).toBeInTheDocument();
        expect(director).toBeInTheDocument();
        expect(genre).toBeInTheDocument();
        expect(runtime).toBeInTheDocument();
        expect(summary).toBeInTheDocument();
    });

    it('renders <Info /> unchanged', () => {
        const { container } = render(<Info />)

        expect(container).toMatchSnapshot()
    });
});
