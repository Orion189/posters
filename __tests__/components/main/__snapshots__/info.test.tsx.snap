// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Info renders <Info /> unchanged 1`] = `
<div>
  <div
    class="cont MuiBox-root css-0"
  >
    <h5
      class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom panelTitle css-h93ljk-MuiTypography-root"
    >
      components.main.Info.title
    </h5>
    <ul
      class="MuiList-root MuiList-padding list css-h4y409-MuiList-root"
    >
      <li
        class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
      >
        <div
          class="MuiListItemText-root MuiListItemText-multiline css-konndc-MuiListItemText-root"
        >
          <span
            class="MuiTypography-root MuiTypography-body1 listItemKey css-10hburv-MuiTypography-root"
          >
            components.main.Info.label.originalTitle
          </span>
          <h6
            class="MuiTypography-root MuiTypography-h6 listItemValue css-cbexrx-MuiTypography-root"
          >
            Test movie title
          </h6>
        </div>
      </li>
      <li
        class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
      >
        <div
          class="MuiListItemText-root MuiListItemText-multiline css-konndc-MuiListItemText-root"
        >
          <span
            class="MuiTypography-root MuiTypography-body1 listItemKey css-10hburv-MuiTypography-root"
          >
            components.main.Info.label.year
          </span>
          <h6
            class="MuiTypography-root MuiTypography-h6 listItemValue css-cbexrx-MuiTypography-root"
          >
            1999
          </h6>
        </div>
      </li>
      <li
        class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
      >
        <div
          class="MuiListItemText-root MuiListItemText-multiline css-konndc-MuiListItemText-root"
        >
          <span
            class="MuiTypography-root MuiTypography-body1 listItemKey css-10hburv-MuiTypography-root"
          >
            components.main.Info.label.country
          </span>
          <h6
            class="MuiTypography-root MuiTypography-h6 listItemValue css-cbexrx-MuiTypography-root"
          >
            US
          </h6>
        </div>
      </li>
      <li
        class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
      >
        <div
          class="MuiListItemText-root MuiListItemText-multiline css-konndc-MuiListItemText-root"
        >
          <span
            class="MuiTypography-root MuiTypography-body1 listItemKey css-10hburv-MuiTypography-root"
          >
            components.main.Info.label.director
          </span>
          <h6
            class="MuiTypography-root MuiTypography-h6 listItemValue css-cbexrx-MuiTypography-root"
          >
            John Doe
          </h6>
        </div>
      </li>
      <li
        class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
      >
        <div
          class="MuiListItemText-root MuiListItemText-multiline css-konndc-MuiListItemText-root"
        >
          <span
            class="MuiTypography-root MuiTypography-body1 listItemKey css-10hburv-MuiTypography-root"
          >
            components.main.Info.label.genre
          </span>
          <h6
            class="MuiTypography-root MuiTypography-h6 listItemValue css-cbexrx-MuiTypography-root"
          >
            Drama
          </h6>
        </div>
      </li>
      <li
        class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
      >
        <div
          class="MuiListItemText-root MuiListItemText-multiline css-konndc-MuiListItemText-root"
        >
          <span
            class="MuiTypography-root MuiTypography-body1 listItemKey css-10hburv-MuiTypography-root"
          >
            components.main.Info.label.runtime
          </span>
          <h6
            class="MuiTypography-root MuiTypography-h6 listItemValue css-cbexrx-MuiTypography-root"
          >
            180
          </h6>
        </div>
      </li>
      <li
        class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
      >
        <div
          class="MuiListItemText-root MuiListItemText-multiline css-konndc-MuiListItemText-root"
        >
          <span
            class="MuiTypography-root MuiTypography-body1 listItemKey css-10hburv-MuiTypography-root"
          >
            components.main.Info.label.summary
          </span>
          <h6
            class="MuiTypography-root MuiTypography-h6 listItemValue css-cbexrx-MuiTypography-root"
          >
            Test movie summary
          </h6>
        </div>
      </li>
    </ul>
  </div>
</div>
`;
